### PR TITLE
Working Search ( but still need some attention )

### PR DESCRIPTION
**Pull request by HelkaPL**
Ok, so far I expand:

getData => now have another function to request from api with queryParams
```
 export const getDataByQuery = async (dataName, currentPage = 1, query) => {
  if (!query) { return; }
  const { data } = await axios.get(
    `${baseUrl}/search/${dataName}?${apiKey}&language=en-US&query=${query}&page=${currentPage}`
  );

  return data;
};
```
movieSaga => added if statement depended on query
    replaced
 `const data = yield call(getPopularData, "movie", currentPage);` 
to:
`const data = yield !query
    ? call(getPopularData, "movie", currentPage)
    : call(getDataByQuery, "movie", currentPage, query);`
but query is geting from payload, but i think it's not a good idea.

Now question is: 

- How, and where set timeout for input query ? ( now its work after every single letter)
- Connect searchresults with pagination ( now its always forward to 2nd page of Popular Request )

### CODE REVIEW

File: src/common/Pagination/index.js [hunk 1]
```diff
@@ -12,34 +12,45 @@ import {
   ButtonText,
 } from "./styled";
 
-const Pagination = ({ location }) => {
+const Pagination = ({ location, fetchedPages }) => {
   const [searchParams, setSearchParams] = useSearchParams();
   const currentPage = Number(searchParams.get("page")) || 1;
-  const totalPages = 500;
+  const query = searchParams.get("search");
+  const totalPages = fetchedPages > 500 ? 500 : fetchedPages;
```
Comment (HelkaPL) [inline src/common/Pagination/index.js:19]: With this part, we will have a correct value of TotalPages :)

File: src/features/movies/moviesSlice.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const moviesSlice = createSlice({
     movies: [],
     genres: [],
     currentPage: 1,
+    totalPage: 500,
```
Comment (HelkaPL) [inline src/features/movies/moviesSlice.js:9]: I'm not sure that value....  mayby  ` totalPages: 0, ` will be better ??

**Comment by HelkaPL** [comment]
How we will be handling changing title when we show searched stuff ?

File: src/features/movies/moviesSaga.js [hunk 3]
```diff
@@ -20,8 +20,10 @@ function* fetchGenresHandler() {
 function* fetchMoviesHandler({ payload }) {
   try {
     const currentPage = payload.currentPage;
-    const data = yield call(getPopularData, "movie", currentPage);
-    yield delay(500);
+    const query = payload.query;
+    const data = yield !query
+      ? call(getPopularData, "movie", currentPage)
+      : call(getDataByQuery, "movie", currentPage, query);
```
Comment (mikolajgraczyk) [inline src/features/movies/moviesSaga.js]: I suggest to not start with negation. Let's remove the `!` and swap these two calls 
`    
const data = yield query
      ? call(getDataByQuery, "movie", currentPage, query)
      : call(getPopularData, "movie", currentPage);
`

Comment (HelkaPL) [reply src/features/movies/moviesSaga.js]: Done.

**Comment by HelkaPL** [comment]
So, now it's ready to be marged, or still need some fix ?

**Review by marcin10lw — APPROVED** [review]
Stunning work!

**Pull request merged**